src/symbols_linux.cpp

```diff
@@ -845,4 +845,39 @@ void Symbols::parseLibraries(CodeCacheArray* array, bool kernel_symbols) {
     }
 }
 
+static bool isValidHandle(const CodeCache* cc, void* handle) {
+    struct link_map* map;
+    // validate that the current loaded library is the same library that was observed during the /proc/self/maps processing
+    if (handle != NULL && dlinfo(handle, RTLD_DI_LINKMAP, &map) == 0) {
+        return cc->imageBase() == (const char*)map->l_addr;
+    }
+    return false;
+}
+
+UnloadProtection::UnloadProtection(const CodeCache *cc) {
+    _lib_handle = NULL;
+    _valid = false;
+
+    if (OS::isMusl() || isMainExecutable(cc->imageBase(), cc->maxAddress()) || isLoader(cc->imageBase())) {
+        _valid = true;
+        return;
+    }
+
+    char* nameCopy = strdup(cc->name());
```

Comment:
Local variables follow `snake_case` naming convention.
Can we avoid `strdup` if there is no `deleted` suffix? Alternatively, we can use a stack-allocated buffer.

Reply:
7a1895a034e241192175b1b5be90c70b83118df6